### ISSUE TITLE
Notify throttling

### ISSUE THREAD

**Issue by ybogdanov** [jwilder/docker-gen#103]
Hi,
It would be nice to have some kind of configurable notification delay. For example, if I run 10 containers in a row within a second, it generates dozens of events and each will cause SIGHUP notify or equivalent. 

With 5 seconds delay, docker-gen may wait and then notify only once.

I can contribute this functionality after you confirm the feature is needed and will be accepted.

**Comment by jwilder** [comment]
Is this causing any specific issues in your case?  Sound similar to #42.

**Comment by ybogdanov** [comment]
@jwilder yes, it's the same. In case of using long running connections, it might be expensive to reload nginx to frequently, because it keeps old worker processes until all connections are closed. It consumes memory and under load may cause OOM kills. Also, depending on the design of upstream servers, it could be expensive to simply cut all connections and let all connections (or part, in case of using multiple balancers) to reconnect.

I can imagine tools that is even more sensitive to reloads/restarts than nginx. So I'm sure we need some king of delay to avoid unnecessary reloads or restarts.

**Comment by jwilder** [comment]
Fixed via #156
